### PR TITLE
Optimized paths for .csv and geojson + extras

### LINKED ISSUE #65: Adjust log level from command-line
Simple --log flag that gets passed transparently to the elasticsearch client

### PR DESCRIPTION
**Pull request by wpears**
Summary of changes:
- Provides three new options for the loader:
     - log: for es log level
     - s_srs: for specifying source spatial reference system on non-spatial datatypes
     - preformatted: flag to indicate GeoJSON that has already been transformed to WGS84 by ogr2ogr

- Allows .csv files to be processed
    - spatial information is auto-detected from columns, based on the header names (x, lon, lon_dd, etc.)
- Various small bug fixes and testing updates

Closes #65, #72, and #73 and supersedes https://github.com/cfpb/grasshopper-loader/pull/69.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,9 +18,12 @@ Data is transformed from these standard formats and loaded into Elasticsearch (t
     - **-t, --transformer** *Default: transformers/[[data file basename]].js* The transformer to use. This converts state-specific data formats to our [Point Data Schema](https://github.com/cfpb/grasshopper/blob/master/docs/point_data_spec.md)
     - **-h, --host** *Default: localhost* The elasticsearch host
     - **-p, --port** *Default: 9200* The elasticsearch port
+    - **-l, --log** *Default: debug* The elasticsearch log level
     - **--index** *Default: address* The elasticsearch index
     - **--type** *Default: point* The elasticsearch type
     - **--profile** *Default: default* The aws credentials profile in ~/.aws/credentials. AWS keys as environment variables will override this setting.')
+    - **--s_srs** *Auto-detects if possible* Source spatial reference system. Needed for untransformed .csv files.
```
Comment (hkeeler) [inline README.md]: `--s_srs` is pretty terse for a long-form option.  Also, I think [dashes are prefered to underscores](http://stackoverflow.com/questions/1253679/should-command-line-options-in-posix-style-operating-systems-be-underscore-style/1253692#1253692). Perhaps `--srs-source` or `--source-srs`?

Comment (hkeeler) [inline README.md]: Also, is there a small set of legit values.  It looks like `WGS84` is mentioned below.  Is it works enumerating which values are supported, or is that common geo knowledge?

Comment (wpears) [inline README.md]: Yeah, I agree. However, -s_srs is the argument that is passed to ogr2ogr. I would be fine having the loader take --source-srs as the parameter, but you don't get the continuity (have to mentally map --source-srs to -s_srs). That said, I agree it is ugly and will change it if you think the continuity between the arguments isn't too important.

Comment (hkeeler) [inline README.md]: Yep, I saw that as soon as I dug down a bit deep.  I guess it depends on who the target audience of this tool is.  If most everyone who'd use this is already familiar with `ogr2ogr` and all its warts, then yes, it does make sense to keep as-is.

File: lib/pointTransformer.js [hunk 2]
```diff
@@ -5,6 +5,64 @@ var formatAddress = require('../lib/formatAddress');
...
+
+}
+
+
+function matchFromArr(props, arr){
+  for(var i=0; i<arr.length; i++){
+    var match = props[arr[i]];
+    if(match) return match;
+  }
+  throw new Error('File resulted in null geometry will no obvious lat/lon fields detected.');
```
Comment (hkeeler) [inline lib/pointTransformer.js]: I think this is supposed to be _...null geometry **with** no obvious..._

File: lib/pointTransformer.js [hunk 3]
```diff
@@ -5,6 +5,64 @@ var formatAddress = require('../lib/formatAddress');
...
+  var lowerCase = {};
+  Object.keys(props).forEach(function(v){
+    lowerCase[v.toLowerCase()] = {key: v, val: props[v]};
+  });
+
+  var xMatch = matchFromArr(lowerCase, lons);
+  var yMatch = matchFromArr(lowerCase, lats);
+  
+  this.coordinates.x = xMatch.key;
+  this.coordinates.y = yMatch.key
```
Comment (hkeeler) [inline lib/pointTransformer.js]: Missing `;`

Comment (wpears) [inline lib/pointTransformer.js]: So I have required semi-colons turned off in my linter and am pretty agnostic about putting them after every statement. That said, the consistency is probably warranted.

File: lib/pointTransformer.js [hunk 4]
```diff
@@ -5,6 +5,64 @@ var formatAddress = require('../lib/formatAddress');
...
+  'long_dd',
+  'lon_x',
+  'long_x'
+];
+
+
+function getGeometry(props, geo){
+  if(geo !== null) return geo;
+
+  if(this.coordinates.x){
```
Comment (hkeeler) [inline lib/pointTransformer.js]: When would `this.coordinates.x` already be set?  Is this just a short circuit so you don't have to redo all the processing again if `getGeometry` has already been called?

**Comment by hkeeler** [comment]
All reviewed up.  Just waiting for Travis to give the :+1: .

File: lib/pointTransformer.js (hunk 4 above)
Comment (wpears) [inline lib/pointTransformer.js]: Yep, since the geometry will need to be pulled out for every row. Turns it from O(n) to O(1)

Comment (hkeeler) [inline lib/pointTransformer.js]: Cool.  Makes sense.

**Pull request merged**